module/os/operation_siren.py

```diff
@@ -116,6 +119,17 @@ def fleet_repair(self, revert=True):
         if revert and prev != self.zone:
             self.globe_goto(prev)
 
+    def need_repair(self, enabled, threshold):
+        if enabled:
+            self.hp_get()
+            check = [round(data, 2) <= threshold if use else False for data, use in zip(self.hp, self.hp_has_ship)]
+            if any(check):
+                logger.info(f'At least one ship below configured threshold {threshold}, retreating to nearest azur port')
+                self.fleet_repair(revert=False)
+                self.hp_reset()
+            else:
+                logger.info('All ships fairly in good condition, continue OS exploration')
+
```

Comment:
Add a new method `handle_fleet_repaire`, use ENABLE_OS_SETTING_REPAIR and OS_SETTING_REPAIR_THRESHOLD directly in it.

Reply:
I don't use those configs directly in code since I had two sets (`ENABLE_OS_SETTING_REPAIR and OS_SETTING_REPAIR_THRESHOLD`) or (`ENABLE_OS_WORLD_REPAIR, OS_WORLD_REPAIR_THRESHOLD`) one for each module.  So based on which method was calling, it would have to change the configs.  Would you rather I reduce this to 1 set and only 1 module capable of configuring this so applies to all of them?

I thought it be more useful to have separate ones per module to have different settings but they can be combined if preferred

Reply:
I think it's enough to use a same value.

Reply:
After thinking more about it and maybe it is just a personal preference.

I'd like to consider keeping as it is if possible. I would think `os_world_clear` since its for clearing unsafe zones which has enemies in larger number. It is likely the threshold would be on the high end so that when its tackling the next zone then it is more likely to be prepared.  But for `os_fully_auto` I would imagine being on the lower end and used after `os_world_clear` in most cases I think it would be preferable doing dailies and farming when they are safe for a smoother run. Saving on changing the configuration too often based on the needs per module.

Also it saves the user from running a functional module and then stopping just to save a setting that affects another module behavior.

`need_repair` has been changed to `handle_fleet_repair', though the content hasn't really changed. But having it accept arguments on how it behaves I think is useful as it allows to use a fixed or configurable threshold. In this case, ensuring the fleet is ready however does not meet the configurable threshold before starting the module can be used as a precaution as exampled in the code.

However if you really don't agree, then I will make the adjustments accordingly for a single paired configuration. The question is which module you'd like the setting to be in `os_world_clear` or `os_fully_auto`? Or perhaps neither and instead added to the `settings` module if that is what you were suggesting

Sorry for the lengthy explanation and I hope it can considered but again I don't mind changing the infrastructure to single pair.

Thanks

Reply:
I want to reduce the number of settings. Alas has much more setthings than other bots for game in 2D. There are too many settings for tiny things which are unneccessary.

When running `os_world_clear`, I assumed that users have perpared for it. They should have 1 or 2 healers in mob fleet. Alas is not responsable to satisfy people who want to attack zones in high CL with weak fleets, but Alas is able to continue running in that situation (Alas keeps turning on auto search and ships keep sacrifising themselves to kill enemies). So I think use the threshold in normal would be enough.

Reply:
Understood, fair point. I shouldn't have problem reducing this is just 1 setting and will be placed in `os_fully_auto` along with the other settings.

Thanks